app/src/main/java/openfoodfacts/github/scrachx/openfood/features/scan/ContinuousScanActivity.kt

```diff
@@ -506,37 +494,23 @@ class ContinuousScanActivity : BaseActivity(), IProductView {
 
         quickViewBehavior.addBottomSheetCallback(bottomSheetCallback)
 
-        beepActive = scannerPrefsRepository.getRingPref()
-        flashActive = scannerPrefsRepository.getFlashPref()
-        autoFocusActive = scannerPrefsRepository.getAutoFocusPref()
-        cameraState = scannerPrefsRepository.getCameraPref().value
+        beepActive = scannerPrefsRepository.ringPref
+        flashActive = scannerPrefsRepository.flashPref
+        autoFocusActive = scannerPrefsRepository.autoFocusPref
+        cameraState = scannerPrefsRepository.cameraPref.value
 
         // Setup barcode scanner
-        if (!useMLScanner) {
-            binding.barcodeScanner.visibility = View.VISIBLE
-            binding.cameraPreviewViewStub.isVisible = false
-            binding.barcodeScanner.barcodeView.decoderFactory = DefaultDecoderFactory(ScannerPreferencesRepository.BARCODE_FORMATS)
-            binding.barcodeScanner.setStatusText(null)
-            binding.barcodeScanner.setOnClickListener {
-                quickViewBehavior.state = BottomSheetBehavior.STATE_HIDDEN
-            }
-            binding.barcodeScanner.barcodeView.cameraSettings.run {
-                requestedCameraId = cameraState
-                isAutoFocusEnabled = autoFocusActive
-            }
 
-            // Start continuous scanner
-            binding.barcodeScanner.decodeContinuous(barcodeScanCallback)
-            beepManager = BeepManager(this)
+        cameraView = if (!scannerPrefsRepository.mlScannerEnabled) {
```

Comment:
(To discuss) Would it be better to have a CameraView factory somewhere else?
So we can use in multiple screens (just in case)